docs/docs/features/backlight.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Backlight
+sidebar_label: Backlight
+---
+
+Backlight is a feature used to control array of LEDs, usually placed through or under switches. Unlike [RGB Underglow](./underglow), backlight currently allows only one color per LED, also LEDs are not addressable, so you can't control individual LEDs.
+
+## Enabling Backlight
+
+To enable backlight on your board or shield, simply enable the `CONFIG_ZMK_BACKLIGHT` configuration values in the `.conf` file of your user config directory as such:
+
+```
+CONFIG_ZMK_BACKLIGHT=y
+```
+
+If your board or shield does not have backlight configured, refer to [Adding Backlight to a Board](#adding-backlight-to-a-board).
+
+## Configuring Backlight
+
+There are various Kconfig options used to configure the backlight feature. These can all be set in the `.conf` file.
+
+| Option                      | Description                                             | Default |
+| --------------------------- | ------------------------------------------------------- | ------- |
+| `ZMK_BACKLIGHT_LEVELS`      | Number of brightness levels                             | 6       |
+| `ZMK_BACKLIGHT_LEVEL_START` | Default brightness level (1 turn off backlight)         | 3       |
+| `ZMK_BACKLIGHT_ON_START`    | Default backlight state                                 | y       |
+| `ZMK_BACKLIGHT_SLEEP`       | Turn off backlight when keyboard goes into sleep status | y       |
+| `ZMK_BACKLIGHT_USB`         | Turn off backlight when USB is disconnected             | n       |
+
+## Adding Backlight to a Board
+
+First, you need to enable PWM by adding the following lines to your `.overlay` file:
+
+```
+&pwm0 {
+	status = "okay";
+	ch0-pin = <45>;
+};
+```
+
+The value `ch0-pin` represents the pin that controls the LEDs. To calculate the value to use, you need a bit of math. You need the hardware port and run it through a function.
+**32 \* X + Y** = `<Pin number>` where X is first part of the hardware port "PX.01" and Y is the second part of the hardware port "P1.Y".
+
+For example, _P1.13_ would give you _32 \* 1 + 13_ = `<45>` and _P0.15_ would give you _32 \* 0 + 15_ = `<15>`.
+
+Then you have to add the following lines to your `.dtsi` file inside the root devicetree node:
+
+```
+backlight: pwmleds {
+    compatible = "pwm-leds";
+    label = "Backlight LEDs";
+    pwm_led_0 {
+        pwms = <&pwm0 45>;
```

Comment:
Is it necessary to put 45 again if it was done in the `&pwm0` definition? Seems odd, but I've never used Zephyr's PWM.

Reply:
It should. A PWM module can have multiple channels, so the meaning of that line is that the led is using the channel which has pin 45.
These things are poorly documented in Zephyr, I found this [blog](https://medium.com/home-wireless/using-a-pwm-device-in-zephyr-7100d089f15c) really useful.